src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -570,4 +570,16 @@ public override IReadOnlyList<Type> GetParameterTypes(string methodName)
             return descriptor.ParameterTypes;
         }
     }
+
+    public class AuthHubConnectionContext : DefaultHubCallerContext
+    {
+        public AuthHubConnectionContext(HubConnectionContext connection, string hubMethodName, object[] hubMethodArguments) : base(connection)
+        {
+            HubMethodName = hubMethodName;
+            HubMethodArguments = hubMethodArguments;
+        }
+
+        public string HubMethodName { get; }
+        public IReadOnlyList<object> HubMethodArguments { get; }
```

Comment:
Open question (which @BrennanConroy and I talked about earlier): Should the arguments be included?

Reply:
The methods will be very useful for cases where the authorization depends on something that is not on the original negotiate request.

Reply:
So far no one has said anything about arguments.

Maybe we shouldn't include them.